itests/curio_test.go

```diff
@@ -65,3 +88,293 @@ func TestCurioNewActor(t *testing.T) {
+	require.GreaterOrEqual(t, len(baseCfg.Addresses), 1)
+
+	require.Contains(t, baseCfg.Addresses[0].MinerAddresses, maddr.String())
+
+	temp := os.TempDir()
+	dir, err := os.MkdirTemp(temp, "curio")
+	require.NoError(t, err)
+	defer func() {
+		_ = os.Remove(dir)
+	}()
+
+	cctx, err := createCliContext(dir)
+	require.NoError(t, err)
+
+	shutdownChan := make(chan struct{})
+	{
+		var ctxclose func()
+		ctx, ctxclose = context.WithCancel(ctx)
+		go func() {
+			<-shutdownChan
+			ctxclose()
+		}()
+	}
+
+	dependencies := &deps.Deps{}
+	dependencies.DB = db
+	dependencies.Full = full
+	seal.SetDevnet(true)
+	err = os.Setenv("CURIO_REPO_PATH", dir)
+	require.NoError(t, err)
+	err = dependencies.PopulateRemainingDeps(ctx, cctx, false)
+	require.NoError(t, err)
+
+	taskEngine, err := tasks.StartTasks(ctx, dependencies)
+	require.NoError(t, err)
+
+	defer taskEngine.GracefullyTerminate()
+
+	dependencies.Cfg.Subsystems.BoostAdapters = []string{fmt.Sprintf("%s:127.0.0.1:32000", maddr)}
+	err = lmrpc.ServeCurioMarketRPCFromConfig(dependencies.DB, dependencies.Full, dependencies.Cfg)
+	require.NoError(t, err)
+
+	go func() {
+		err = rpc.ListenAndServe(ctx, dependencies, shutdownChan) // Monitor for shutdown.
+		require.NoError(t, err)
+	}()
+
+	finishCh := node.MonitorShutdown(shutdownChan)
+
+	mid, err := address.IDFromAddress(maddr)
+	require.NoError(t, err)
+
+	mi, err := full.StateMinerInfo(ctx, maddr, types.EmptyTSK)
+	require.NoError(t, err)
+
+	nv, err := full.StateNetworkVersion(ctx, types.EmptyTSK)
+	require.NoError(t, err)
+
+	wpt := mi.WindowPoStProofType
+	spt, err := miner2.PreferredSealProofTypeFromWindowPoStType(nv, wpt, false)
+	require.NoError(t, err)
+
+	var machines []string
+	err = db.Select(ctx, &machines, `select host_and_port from harmony_machines`)
+	require.NoError(t, err)
+
+	require.Len(t, machines, 1)
+	laddr, err := net.ResolveTCPAddr("tcp", machines[0])
+	require.NoError(t, err)
+
+	ma, err := manet.FromNetAddr(laddr)
+	require.NoError(t, err)
+
+	var apiToken []byte
+	{
+		type jwtPayload struct {
+			Allow []auth.Permission
+		}
+
+		p := jwtPayload{
+			Allow: api.AllPermissions,
+		}
+
+		sk, err := base64.StdEncoding.DecodeString(storageSecret)
+		require.NoError(t, err)
+
+		apiToken, err = jwt.Sign(&p, jwt.NewHS256(sk))
+		require.NoError(t, err)
+	}
+
+	ctoken := fmt.Sprintf("%s:%s", string(apiToken), ma)
+	err = os.Setenv("CURIO_API_INFO", ctoken)
+	require.NoError(t, err)
+
+	capi, ccloser, err := rpc.GetCurioAPI(&cli.Context{})
+	require.NoError(t, err)
+	defer ccloser()
+
+	scfg := storiface.LocalStorageMeta{
+		ID:         storiface.ID(uuid.New().String()),
+		Weight:     10,
+		CanSeal:    true,
+		CanStore:   true,
+		MaxStorage: 0,
+		Groups:     []string{},
+		AllowTo:    []string{},
+	}
+
+	err = capi.StorageInit(ctx, dir, scfg)
+	require.NoError(t, err)
+
+	err = capi.StorageAddLocal(ctx, dir)
+	require.NoError(t, err)
+
+	err = capi.LogSetLevel(ctx, "harmonytask", "DEBUG")
```

Comment:
This whole top part should probably be a `ConstructCurioTest` function, or something of this sort; Amazing to finally have at least one real integration test for the sealing pipelin!